packages/EDA/src/svm.ts

```diff
@@ -494,23 +473,30 @@ function getUnpackedModel(packedModel: any): any {
 export async function getPrediction(df: DG.DataFrame, packedModel: any): Promise<DG.DataFrame> {
   const model = getUnpackedModel(new Uint8Array(packedModel));
 
+  console.log(model);
+
   const resNumeric = await predict(model, df.columns);
   const res = DG.Column.string(PREDICTION, resNumeric.length);
   const categories = model.realLabels.categories;
-  for (var i = 0; i < res.length; i++) {
+  for (let i = 0; i < res.length; i++)
     res.set(i, resNumeric.get(i) == -1 ? categories[0] : categories[1]);
-  }
+
 
   return DG.DataFrame.fromColumns([res]);
 } // getPrediction
 
-
 export function isApplicableSVM(df: DG.DataFrame, predictColumn: string): boolean {
   const columns = df.columns;
   const labels = columns.byName(predictColumn);
   columns.remove(predictColumn);
-  var res: boolean = labels.type == 'string';
-  for (var i = 0; i < columns.length; i++)
-    res = res && (columns.byIndex(i).type == 'double' || columns.byIndex(i).type == 'int');
-  return res;
-}
\ No newline at end of file
+
+  if ((labels.type !== DG.COLUMN_TYPE.STRING) || (labels.categories.length > 2))
+    return false;
+
+  for (const col of columns) {
+    if ((col.type !== DG.COLUMN_TYPE.FLOAT) && (col.type !== DG.COLUMN_TYPE.INT))
```

Comment:
QNUM and BIGINT are not handled. I've seen it before somewhere, too. Let's use column.isNumerical?

Reply:
Fixed